### ISSUE TITLE
Issue related to setReportInterval()

### ISSUE THREAD

**Issue by babbarkrishan** [aws/aws-iot-device-sdk-java#100]
Hi,

I am working on AWS IoT and looking for event based reporting. 
As per below issue, it seems AWS IoT SDK still does not support event based reporting.
https://github.com/aws/aws-iot-device-sdk-java/issues/10

Please suggest if you are planning to implement it? 
I have a similar requirement i.e. I have a Java Application having AWS IoT SDK. Device connecting to Java Application and sending periodic packets/messages (may be after every 15 seconds). Sometimes due to network issue device may send packet after may be half an hour or so.
I want to publish data to AWS IoT Shadow only after getting packets from device. Where as setReportInterval() will keep publishing data on regular interval irrespective of device sending data or not.

I am using following version SDK version
aws-iot-device-sdk-java - 1.3.0 (Maven Dependency)

Another issue, i am facing is with setReportInterval().
I tried using "awsThing.setReportInterval(5000);".
It is working but it is not working as expected or explained.
E.g. in case of 5000 value, shadow is being updated every 10 second whereas my device was sending message every 5 seconds.
Also, in case of 10000 value (awsThing.setReportInterval(10000);) and device sending message every 5 seconds, i am getting error "Connection temporarily lost" and it is not updating shadow at all. Please see below logs.

Logs
  : Connection successfully established
2019-05-17 23:22:30.344  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AbstractAwsIotClient         : Client connection active: BT_PUBLISHER
2019-05-17 23:22:34.966  INFO 15272 --- [_Task_Executor4] i.b.i.d.s.i.MsgParserService             : Actual message: @P,0DB3,125,410,356611071222994,1557438361,1557472194,1557472194,151139120,-33794205,141,2,13,990,1,0,0,0,,2000,2000,,7,7,%AN1%AN2%CI
2019-05-17 23:22:34.971  INFO 15272 --- [_Task_Executor4] i.b.i.d.p.SeperatorParser                : io.boral.iot.devicebrokerpublisher.entity.AK11Entity@7ecb4a80
2019-05-17 23:22:34.972  INFO 15272 --- [_Task_Executor4] i.b.i.d.s.i.MsgParserService             : Generic Entity: io.boral.iot.devicebrokerpublisher.entity.AK11Entity@7ecb4a80
2019-05-17 23:22:34.972 DEBUG 15272 --- [_Task_Executor4] i.b.i.d.s.i.MsgParserService             : thingName: BT_AK11_356611071222994 found in map.
2019-05-17 23:22:34.972  INFO 15272 --- [_Task_Executor4] i.b.i.d.s.i.MsgParserService             : Device has been updated successfully. Thing name: BT_AK11_356611071222994
2019-05-17 23:22:37.521  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AwsIotConnection             : Connection temporarily lost
2019-05-17 23:22:37.528  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AbstractAwsIotClient         : Client connection lost: BT_PUBLISHER
2019-05-17 23:22:39.972 ERROR 15272 --- [           main] i.b.i.d.s.i.QueueServiceImpl             : Processing test data completed.
2019-05-17 23:22:40.538  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AwsIotConnection             : Connection is being retried
2019-05-17 23:22:40.773  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AwsIotConnection             : Connection successfully established
2019-05-17 23:22:40.773  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AbstractAwsIotClient         : Client connection active: BT_PUBLISHER
2019-05-17 23:22:47.953  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AwsIotConnection             : Connection temporarily lost
2019-05-17 23:22:47.956  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AbstractAwsIotClient         : Client connection lost: BT_PUBLISHER
2019-05-17 23:22:50.955  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AwsIotConnection             : Connection is being retried
2019-05-17 23:22:51.140  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AwsIotConnection             : Connection successfully established
2019-05-17 23:22:51.140  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AbstractAwsIotClient         : Client connection active: BT_PUBLISHER
2019-05-17 23:22:58.347  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AwsIotConnection             : Connection temporarily lost
2019-05-17 23:22:58.349  INFO 15272 --- [pool-2-thread-1] c.a.s.i.c.c.AbstractAwsIotClient         : Client connection lost: BT_PUBLISHER

Regards,
Krishan

**Comment by justinboswell** [comment]
Hello!

Event based reporting is not something we support, or will be adding to this version of the SDK. However, I have added it to the roadmap for the [v2 IoT Java Device SDK](https://github.com/awslabs/aws-iot-device-sdk-java-v2), which is in preview. I can't say whether or not this will make it before the v2 SDK hits 1.0, but this seems like a completely reasonable feature to support.
